Description: Fix build problem on archs with signed char
 The return type of getc() is int to be able to return both characters
 and EOF (-1).  The test in the code is always false when the return is
 stored in a char when char is unsigned.  This fixes build problem on
 arm* and many other architectures.
Author: Petter Reinholdtsen <pere@debian.org>
Bug-Debian: https://bugs.debian.org/789248
Forwarded: no
Reviewed-By: Petter Reinholdtsen <pere@debian.org>
Last-Update: 2015-06-20

--- bs1770gain-0.4.3.orig/libffsox-2/ffsox_csv2avdict.c
+++ bs1770gain-0.4.3/libffsox-2/ffsox_csv2avdict.c
@@ -255,11 +255,14 @@ static int priv_get_utf8(priv_t *b)
   static unsigned short mask[] = {192, 224, 240}; 
 
   char *wp=b->ch;
+  int firstbyte;
   size_t n; 
 
   // read first byte into buffer
-  if (EOF==(*wp=getc(b->f)))
+  firstbyte = getc(b->f);
+  if (EOF==firstbyte)
     goto error;
+  *wp = (char)firstbyte;
 
   // check how many more bytes need to be read for character
   n = 0;
